### PR TITLE
add ustring builtin type (unsigned char*)

### PR DESCRIPTION
**Pull request by pgu-swir**
This answers issue #124 by using "ustring" and casting string types to (const char *) when used in the codecWrite. 

This is still WIP as the server shim code needs as well to do some cast... but I wanted to know if the direction is satisfactory for you

### CODE REVIEW

**Comment by MichalPrincNXP** [comment]
Hello @pgu-swir , I am sorry not being able to respond sooner when discussed with @Hadatko about this problem. To be honest, @Hadatko is more experienced in this field than me, however it seems to me OK to introduce ustring type and also the so-far-done steps should be ok. As for the failing Travis, please ignore this in both of your PRs ... I have migrated erpc project from travis-ci.org to travis-ci.com recently and there are build issues since that time (head of develop branch also fails under travis-ci.com while it was passing under travis-ci.org). Will try to solve this soon.

**Comment by pgu-swir** [comment]
Hi @MichalPrincNXP , no worries. The "full" patch is available now and works for me and seems to have passed Travis tests :-)

**Comment by pgu-swir** [comment]
Anything else I should do?

File: erpcgen/src/templates/c_coders.template [hunk 1]
```diff
@@ -232,7 +232,7 @@ codec->readData({$info.name}, {$info.sizeTemp} * sizeof({$info.builtinTypeName})
 {# Encode sending data #}
 {% def encodeBuiltinType(info) -----------------  %}
 {% if info.builtinType == "kStringType" %}
-codec->writeString(strlen({$info.name}), {$info.name});
+codec->writeString(strlen((const char*){$info.name}), (const char*){$info.name});
```
Comment (Hadatko) [inline erpcgen/src/templates/c_coders.template:235]: if info.builtinType == "kStringType" looks like it is not counting with ustring type. Am i wrong?

Comment (Hadatko) [reply erpcgen/src/templates/c_coders.template:235]: Ok i looked into code and it should be ok.

**Review by Hadatko — APPROVED** [review]
We need to fix tests. If they are passing, then i think so far so good ;)

**Review by MichalPrincNXP — APPROVED** [review]
Hello @pgu-swir , I have just finished testing on my side and all seems to be OK. Thanks for this effort!

**Comment by Hadatko** [comment]
@pgu-swir it would be good if you could add some erpcgen pytest. Not execution but for generating ouput. To keep support for future. Take a look here: https://github.com/EmbeddedRPC/erpc/tree/develop/erpcgen/test

**Comment by MichalPrincNXP** [comment]
Well, Dusan, I have already updated tests + added one more test on my side ... I will provide this update on the develop branch later this week with other NXP changes for 1.8.0 version.

**Comment by pgu-swir** [comment]
> @pgu-swir it would be good if you could add some erpcgen pytest. Not execution but for generating ouput. To keep support for future. Take a look here: https://github.com/EmbeddedRPC/erpc/tree/develop/erpcgen/test
>
I'll look at it - I need to figure out exactly what/where to add something as ustring are really a string

**Comment by MichalPrincNXP** [comment]
> 
> 
> > @pgu-swir it would be good if you could add some erpcgen pytest. Not execution but for generating ouput. To keep support for future. Take a look here: https://github.com/EmbeddedRPC/erpc/tree/develop/erpcgen/test
> 
> I'll look at it - I need to figure out exactly what/where to add something as ustring are really a string

Do not worry about this, I will manage that as written in my previous note.

**Comment by pgu-swir** [comment]
> > > @pgu-swir it would be good if you could add some erpcgen pytest. Not execution but for generating ouput. To keep support for future. Take a look here: https://github.com/EmbeddedRPC/erpc/tree/develop/erpcgen/test
> > 
> > 
> > I'll look at it - I need to figure out exactly what/where to add something as ustring are really a string
> 
> Do not worry about this, I will manage that as written in my previous note.
>
oh, thanks so much!

**Pull request merged**